std/fs/empty_dir_test.ts

```diff
@@ -123,3 +123,116 @@ test(function emptyDirSyncIfItExist(): void {
     Deno.removeSync(testDir, { recursive: true });
   }
 });
+
+test(async function emptyDirPermission(): Promise<void> {
+  interface Scenes {
+    read: boolean; // --allow-read
+    write: boolean; // --allow-write
+    async: boolean;
+    output: string;
+  }
+
+  const testfolder = path.join(testdataDir, "testfolder");
+
+  await Deno.mkdir(testfolder);
+
+  await Deno.writeFile(
+    path.join(testfolder, "child.txt"),
+    new TextEncoder().encode("hello world")
+  );
+
+  const scenes: Scenes[] = [
+    // 1
+    {
+      read: false,
+      write: false,
+      async: true,
+      output: "run again with the --allow-read flag"
+    },
+    {
+      read: false,
+      write: false,
+      async: false,
+      output: "run again with the --allow-read flag"
+    },
+    // 2
+    {
+      read: true,
+      write: false,
+      async: true,
+      output: "run again with the --allow-write flag"
+    },
+    {
+      read: true,
+      write: false,
+      async: false,
+      output: "run again with the --allow-write flag"
+    },
+    // 3
+    {
+      read: false,
+      write: true,
+      async: true,
+      output: "run again with the --allow-read flag"
+    },
+    {
+      read: false,
+      write: true,
+      async: false,
+      output: "run again with the --allow-read flag"
+    },
+    // 4
+    {
+      read: true,
+      write: true,
+      async: true,
+      output: "success"
+    },
+    {
+      read: true,
+      write: true,
+      async: false,
+      output: "success"
+    }
+  ];
+
+  try {
+    for (const s of scenes) {
+      console.log(
+        `test ${s.async ? "emptyDir" : "emptyDirSync"}("testdata/testfolder") ${
+          s.read ? "with" : "without"
+        } --allow-read & ${s.write ? "with" : "without"} --allow-write`
+      );
+
+      const args = [Deno.execPath(), "run"];
+
+      if (s.read) {
+        args.push("--allow-read");
+      }
+
+      if (s.write) {
+        args.push("--allow-write");
+      }
+
+      args.push(
+        path.join(testdataDir, s.async ? "empty_dir.ts" : "empty_dir_sync.ts")
+      );
+      args.push("testfolder");
+
+      const { stdout } = Deno.run({
+        stdout: "piped",
+        cwd: testdataDir,
+        args: args
+      });
```

Comment:
I’m concerned starting all these subprocesses is going to slow down the tests. I wish it could be tested another way.

Reply:
I also think this will slow down the test

But it seems that there is no effective way to solve this problem.

Because when we run the tests of the standard library with `--allow-all` flag

We need to test how these functions behave when we don't yet have permissions.

Then there are only two solution:

1. Dynamically change current permissions (add / delete)
2. Start a child process